### PR TITLE
Vendor `gardener@1.81.7`, Upgrade dependencies and implement `ForceDelete` method for extensions

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/kind enhancement
/area delivery open-source
/platform equinix-metal

**What this PR does / why we need it**:
- Vendor `gardener@1.81.7`
- Implement `ForceDelete` method for extensions
- The force deletion will only work if the Shoot has a user error code, See https://github.com/gardener/gardener/blob/master/docs/usage/shoot_operations.md#force-deletion, which is absent in this extension as of today. 

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/3110

**Special notes for your reviewer**:

**Release note**:

```feature user
The following dependencies are updated:
- `gardener/gardener`: v1.80.3 -> v1.81.7
```

### CODE REVIEW

**Review by ary1992 — COMMENTED** [review]
/lgtm

**Review by acumino — APPROVED** [review]
/lgtm

**Comment by shafeeqes** [comment]
@schrodit Can we merge this?

File: pkg/controller/infrastructure/actuator_delete.go [hunk 1]
```diff
@@ -50,3 +50,7 @@ func (a *actuator) Delete(ctx context.Context, log logr.Logger, infrastructure *
 		SetEnvVars(generateTerraformInfraVariablesEnvironment(infrastructure.Spec.SecretRef)...).
 		Destroy(ctx)
 }
+
+func (a *actuator) ForceDelete(_ context.Context, _ logr.Logger, _ *extensionsv1alpha1.Infrastructure, _ *extensionscontroller.Cluster) error {
+	return nil
+}
```
Comment (schrodit) [inline pkg/controller/infrastructure/actuator_delete.go:56]: what is the expected behavior to actually support ForceDelete?
e.g. the gcp extension is waiting for the managed resource deletion https://github.com/gardener/gardener-extension-provider-gcp/pull/659/files#diff-c2e01c540745a0df87bb8d3b604303079a1f0b70aa33e17b649dd0448fa54f75

Should we check here for resources that might leak external infra?

Comment (rfranzke) [reply pkg/controller/infrastructure/actuator_delete.go:56]: ping @shafeeqes

Comment (shafeeqes) [reply pkg/controller/infrastructure/actuator_delete.go:56]: > what is the expected behavior to actually support ForceDelete? e.g. the gcp extension is waiting for the managed resource deletion https://github.com/gardener/gardener-extension-provider-gcp/pull/659/files#diff-c2e01c540745a0df87bb8d3b604303079a1f0b70aa33e17b649dd0448fa54f75
> 
> Should we check here for resources that might leak external infra?

👀 
The gcp extension also returns nil for infrastructure delete.
Without valid credentials, we cannot clean them up, that's why we return nil.

**Review by rfranzke — APPROVED** [review]

**Pull request merged**